data/data/aws/master/main.tf

```diff
@@ -59,53 +59,42 @@ resource "aws_iam_role_policy" "master_policy" {
   "Statement": [
     {
       "Action": [
-        "ec2:DescribeInstances",
-        "ec2:DescribeRegions",
-        "ec2:DescribeRouteTables",
-        "ec2:DescribeSecurityGroups",
-        "ec2:DescribeSubnets",
-        "ec2:DescribeVolumes",
+        "ec2:AttachVolume",
+        "ec2:AuthorizeSecurityGroupIngress",
         "ec2:CreateSecurityGroup",
         "ec2:CreateTags",
         "ec2:CreateVolume",
-        "ec2:ModifyInstanceAttribute",
-        "ec2:ModifyVolume",
-        "ec2:AttachVolume",
-        "ec2:AuthorizeSecurityGroupIngress",
         "ec2:DeleteSecurityGroup",
         "ec2:DeleteVolume",
+        "ec2:Describe*",
```

Comment:
this change is in stark contrast to the precedent we wanted to set with https://github.com/openshift/installer/pull/3451
now we have a expanding IAM permission even though it is Describe*, and no longer a smallest list.

I want to understand, we are we going back a little on our stance here, and maybe we can update the https://github.com/openshift/installer/blob/master/docs/dev/aws/iam_permissions.md to clarify our loosened requirements.

Reply:
We can be explicit about the Describe list as it is now; it's read-only operations. The main thing this does is scope the changes the master nodes can make to storage, loadbalancers or the environment which is the goal. This allows fixes to be made to this and subsequent versions where an extra read/describe on a resource may be needed to provide an accurate fix for public/private clusters without then bumping into a permission required to *see* something.

I still like the words in the README, this just doesn't block a fix or smarter controller/operator due to an AWS discoverability/visibility issue.